framework/contracts/account/src/contract.rs

```diff
@@ -405,6 +417,62 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
     }
 }
 
+#[cfg_attr(feature = "export", cosmwasm_std::entry_point)]
+pub fn ibc_source_callback(
+    deps: DepsMut,
+    env: Env,
+    msg: IbcSourceCallbackMsg,
+) -> StdResult<IbcBasicResponse> {
+    match msg {
+        IbcSourceCallbackMsg::Acknowledgement(IbcAckCallbackMsg {
+            acknowledgement,
+            original_packet,
+            relayer: _,
+            ..
+        }) => {
+            let packet_identifier = ICS20PacketIdentifier {
+                channel_id: original_packet.src.channel_id,
+                sequence: original_packet.sequence,
+            };
+
+            let (outcome, stored_msgs) =
+                if acknowledgement.data == StdAck::success(b"\x01").to_binary() {
+                    (
+                        "result",
+                        ICS20_CALLBACKS
+                            .load(deps.storage, packet_identifier.clone())?
+                            .into_iter()
+                            .map(|msg| wasm_execute(&env.contract.address, &msg, vec![]))
+                            .collect::<Result<Vec<_>, _>>()?,
+                    )
```

Comment:
Why not to execute `execute` function instead of calling execute endpoint? You can fill MessageInfo with blank sender, this way we are fine if account is the admin of the account